fabric-connector/src/main/java/com/luxoft/fabric/FabricConfig.java

```diff
@@ -368,46 +367,42 @@ public Channel getChannel(HFClient hfClient, String channelName, User fabricUser
 
     private Channel.PeerOptions getPeerOptionsWithRoles(String channelName, String peerName, Channel.PeerOptions commonPeerOptions) {
 
-        //Add all roles by default except ServiceDiscovery in line with NetworkConfig behaviour
-        EnumSet<Peer.PeerRole> peerRoles = EnumSet.of(Peer.PeerRole.CHAINCODE_QUERY,
-                Peer.PeerRole.ENDORSING_PEER,
-                Peer.PeerRole.LEDGER_QUERY,
-                Peer.PeerRole.EVENT_SOURCE);
+        requireNonNull(getRoot().channels.get(channelName), "No channel with name " + channelName);
```

Comment:
Why not assign intermediate results and use it in the following statements?

Reply:
Done